core/src/main/java/org/opensearch/sql/expression/function/QueryContext.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.expression.function;
+
+import java.time.Clock;
+
+/**
+ * Class to capture values that may be necessary to implement some functions.
+ * An example would be query execution start time to implement now().
+ */
+public interface QueryContext {
```

Comment:
I feel FunctionParameter is more concrete name.

Reply:
I feel "function" is overloaded and we don't want to limit this to just functions.  There's no reason why the context couldn't be passed to non-function expressions, if for example, we want to save the storage mechanism in here (catalog vs SQL vs something else) and have that available to... say... identifiers or datatypes.